pkg/3scale/amp/operator/upgrade.go

```diff
@@ -581,3 +589,30 @@ func (u *UpgradeApiManager) findTagReference(tagRefName string, tagRefs []imagev
 	}
 	return -1
 }
+
+func (u *UpgradeApiManager) upgradeSystemMasterApicastSecret() (reconcile.Result, error) {
+	existing := &v1.Secret{}
+	secretNamespacedName := types.NamespacedName{
+		Name:      component.SystemSecretSystemMasterApicastSecretName,
+		Namespace: u.Cr.Namespace,
+	}
+	err := u.Client.Get(context.TODO(), secretNamespacedName, existing)
+	// NotFound also regarded as error, as secret is expected to exist
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if _, ok := existing.Data["BASE_URL"]; ok {
```

Comment:
JSON patch is not really needed in this case.

In Go you can delete a map key. Go allows you to use:
```
delete(mymap, mymapkey)
```
The delete function doesn't return anything, and will do nothing if the specified key doesn't exist.

So you can check if the key exists or not and if it does delete it and update the object.

If you still want to use JSON patch I'm ok with it too as it does the work.